doc/adrs/187_add_visibility_tool_on_diagram_nodes.adoc

```diff
@@ -0,0 +1,87 @@
+= ADR-187 - Add visibility tool on diagram nodes
+
+== Context
+
+The Show/Hide menu of the diagram nodes context menu (a.k.a the palette) is difficult to understand for the end-users.
+Furthermore, it forces users to click 3 times to hide a graphical node:
+- right click on the diagram node to open the palette
+- left click on the Show/Hide section
+- click on Hide menu item
+
+== Decision
+
+A new UI should be available in Sirius Web, allowing to easily manage the visibility of diagram nodes.
+This UI would be an icon and would be placed on the top right corner of the node.
+The specifier will tell which node descriptions will have this UI.
+
+=== Backend
+
+==== View DSL Diagram
+
+On a `diagram::NodeDescription`, it will be possible to create a new `diagram::NodeAction`.
+The `diagram::NodeAction` will have three properties:
+- a labelExpression, to display the label of the action as a tooltip
+- a preconditionExpression, to only make available the action on a specific condition
+- an iconURLsExpression, to display the icon of the action
+On a `diagram::NodeAction`, it will be possible to add the view dsl operations (e.g. change context, create instance, delete element...)
+
+When a new `diagram::NodeDescription` will be created, a default `diagram::NodeAction` will be created at the same time (as it is already the case for the node palette).
+This default `diagram::NodeAction` will allow to automatically hide the node.
+
+==== API
+
+A new API will allow to call a `diagramNode#Action` declared on frontend side (see Frontend section in this document for more details).
+
+.org.eclipse.sirius.components.diagram.actions.api
+[code,java]
+----
+public interface INodeActionService {
+  void callNodeAction(String id);
+};
+----
+
+This new API could be called by a specifier in the View DSL Diagram, on a `view::ChangeContext` (e.g. `aql:self.callNodeAction('siriusWeb_diagramNode#Action_manageVisibilityMenu')`).
```

Comment:
Why? It's like a tool, the specifier can use the model operation DSL to provide the behavior and we will call that. This feels like the complete opposite of everything we do.

Reply:
Done